board.py

```diff
@@ -1,33 +1,90 @@
 class Board:
-    board_state = []
+    def __init__(self):
+        self.board_state = []
 
     def new_game(self, numrows=3):
         board_area = numrows*numrows
-        new_board = [" "] * board_area
+        new_board = [""] * board_area
         self.board_state = new_board
 
-def spot_exists(current_state, spotChoice):
-    if (spotChoice - 1) >= 0 and spotChoice <= len(current_state):
-        return True
-    else:
-        return False
+class WinningCombos:
+    def __init__(self, size):
+        self.size = size
+        self.winning_combos = []
+
+    def win_by_row(self):
+        winning_rows = []
+        single_row = []
+        for spot in range(0, self.size*self.size):
+            single_row.append(spot)
+            if len(single_row) == self.size:
+                winning_rows.append(single_row)
+                single_row = []
+        return winning_rows
+
+    def win_by_column(self, rows):
+        winning_columns = []
+        for outer in range(0, self.size):
+            single_column = []
+            for inner in range(0, self.size):
+                single_column.append(rows[inner][outer])
+            winning_columns.append(single_column)
+        return winning_columns
 
-def is_occupied(current_state, spot):
-    if current_state[spot - 1] != " ":
-        return True
-    else:
-        return False 
+    def diagonal_left_to_right_win(self, rows):
+        left_to_right = []
+        for spot in range(0, self.size):
+            left_to_right.append(rows[spot][spot])
+        return left_to_right
+    
+    def diagonal_right_to_left_win(self, rows):
+        right_to_left = []
+        size = self.size
+        spot = size
+        while spot > 0:
+            right_to_left.append(rows[spot - 1][size - spot])
+            spot -= 1
+        right_to_left.reverse()
+        return right_to_left
+    
+    def create_winning_combos(self):        
```

Comment:
It seems odd to me to build a list of all possible winning conditions and check against them. Have you thought of other strategies for verifying a winner?

Reply:
I thought about having logic that checked to see if there were three consecutive symbols at any location, but realized I would still need the board spots to compare against. This approach seemed more through.

Reply:
So what's the difference between this class and a dictionary that holds all the winning combinations for a given board size? Would you ever want to have that dictionary in the first place?